### PR TITLE
Refactor project

### PR DESCRIPTION
**Pull request by kathrynkodama**
See https://github.com/kathrynkodama/lsp4jakarta/tree/decouple-lsp4jakarta-jdt

Refactor project into 3 main components
- jakarta.ls
- jakarta.jdt - Eclipse JDT LS extension, built with Maven and Tycho
- jakarta.eclipse - Jakarta EE Eclipse client plug-in, built with Maven and Tycho

- Update documentation to include more detailed steps on how to build the project locally and set up eclipse workspace.

Limitations:
- current plug-ins produced by Maven and Tycho for `jakarta.jdt` and `jakarta.eclipse` do not work when locally installed to a new Eclipse instance. The only way to run the application is through setting up an Eclipse workspace and running through Eclipse.

### CODE REVIEW

File: docs/BUILDING.md [hunk 1]
```diff
@@ -2,31 +2,67 @@
 
 ## Prerequisites
 
-[JavaSE-11](https://www.oracle.com/ca-en/java/technologies/javase-jdk11-downloads.html), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) are required to build `lsp4jakarta` and `jakarta-eclipse`.
+[JavaSE-11](https://www.oracle.com/ca-en/java/technologies/javase-jdk11-downloads.html), [Maven](https://maven.apache.org/download.cgi), and [Eclipse](https://www.eclipse.org/downloads/) (Eclipse IDE for Enterprise Java and Web Developers is recommended) are required to build the lsp4jakarta project.
```
Comment (ericglau) [inline docs/BUILDING.md:5]: For clarity, are these requirements for this entire LSP4Jakarta project (i.e. all three of jakarta.ls, jakarta.jdt, jakarta.eclipse)?

File: docs/BUILDING.md [hunk 2]
```diff
@@ -2,31 +2,67 @@
...
+The following are instructions to set up your Eclipse workspace
 
-3. Import `org.eclipse.lsp4jakarta.core`, `org.eclipse.lsp4jakarta.tests` and `lsp4jakarta` projects in Eclipse (File --> Import --> General --> Projects from  Folder or Archive --> Select your LSP4Jakarta clone)
+1. Clone this repository onto your local machine
 
-4. Ensure that projects are being built with `JavaSE-11` ("Right-click project" --> "Properties" --> "Java Build Path" --> "Libraries")
+    `git clone git@github.com:eclipse/lsp4jakarta.git`
 
-5. Run the `./buildAll.sh` script to build the `lsp4jakarta-1.0-SNAPSHOT-jar-with-dependencies.jar`. This script also copies the `lsp4jakarta-1.0-SNAPSHOT-jar-with-dependencies.jar` to the `/jakarta-eclipse/org.eclipse.lsp4jakarta.core` directory 
+2. Build the project with Maven by running the `./buildAll.sh` script from the `lsp4jakarta` directory with Maven and [Tycho](https://github.com/eclipse/tycho). This script runs `mvn clean install` on the components the following order:
```
Comment (ericglau) [inline docs/BUILDING.md:16]: `on the components in the following order`

File: docs/BUILDING.md [hunk 3]
```diff
@@ -2,31 +2,67 @@
...
+
+    <img src="/docs/images/building_tycho_configurator_1.png" alt="Tycho configurator errors" height="60%" width="60%"/>
+    
+    <img src="/docs/images/building_tycho_configurator_2.png" alt="Discover m2e connectors" height="40%" width="40%"/> <img src="/docs/images/building_tycho_configurator_3.png" alt="Installing Tycho Project Configurators" height="50%" width="50%"/>
+
+4. Ensure that the Java projects are being built with `JavaSE-11` ("Right-click project" --> "Properties" --> "Java Build Path" --> "Libraries")
+
+5. Configure the Java build path for the `org.eclipse.lsp4jakarta.lsp4e.core` project:
+
+    1. "Right-click org.eclipse.lsp4jakarta.lsp4e.core project" --> "Properties" --> "Java Build Path" --> "Libraries"
```
Comment (ericglau) [inline docs/BUILDING.md:38]: `"Right-click org` - `Right-click "org`

File: README.md [hunk 4]
```diff
@@ -5,49 +5,25 @@ The Eclipse LSP4Jakarta (Language Server for Jakarta EE) project provides core l
 
 This project contains: 
 
-- [lsp4jakarta](/lsp4jakarta) - Language Server for Jakarta EE
-- [jakarta-eclipse](/jakarta-eclipse) - Eclipse JDT LS extension and Eclipse IDE client that consumes the Language Server for Jakarta EE
+- [jakarta.ls](/jakarta.ls) - Language Server for Jakarta EE
+- [jakarta.jdt](/jakarta.jdt) - Jakarta EE Eclipse JDT LS extension 
+- [jakarta.eclipse](/jakarta-eclipse) - Eclipse IDE client that consumes the Language Server for Jakarta EE
```
Comment (ericglau) [inline README.md:10]: The link is broken, should be `(/jakarta.eclipse)`

File: docs/BUILDING.md (hunk 1 above)
Comment (kathrynkodama) [reply docs/BUILDING.md:5]: Right now, the only way to run the project is through Eclipse so users would need all of these for all three components

**Pull request merged**